catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java

```diff
@@ -44,10 +50,13 @@ public FeedRepository(CollectionDAO dao) {
   }
 
   @Transaction
-  public Thread create(Thread thread) throws IOException, ParseException {
-    // Validate user creating thread
+  public Thread create(Thread thread, SecurityContext securityContext) throws IOException, ParseException {
     String fromUser = thread.getPosts().get(0).getFrom();
-    dao.userDAO().findEntityByName(fromUser);
+
+    if (SecurityUtil.isSecurityEnabled(securityContext)) {
+      // Validate user creating thread if security is enabled
```

Comment:
Is this intended to throw exception when security is enabled and user does not exist?

Reply:
Hi @mithmatt,

Yes, the intention is to not allow the creation of threads by a non-existent user. We can improve the exception handling here and throw a 400 with a message. I will take care of it in my next patch.

Reply:
We should probably move this to FeedResource.java, if that's the case

(to follow the pattern where all security related auth is handled at the API layer)

Reply:
In this case, the createThread method is also called directly to create automated threads based on change events (https://github.com/open-metadata/OpenMetadata/pull/2814/files#diff-03713e3b532bdecbafc71ae905d87f87a1ce1b81d7478adb34e61076ae442948R78). If we move the auth check to resource, then automated threads will be able to bypass this check.  But, I am ok with either way because the change events must have a valid user making the change (which we extract from the logged-in User Principal), so the chances of this user not existing in the DB is almost 0.

Reply:
@vivekratnavel what you mean non existent user? incase anonymous user?

Reply:
@vivekratnavel this kind of check means anyone who is trying out OpenMetadata using docker will not be able to get any threads. We should allow users to test and restrict this only for security enabled OpenMetadata. 
We discussed this previously
1. Create a anonymous user as part of NoOp user
2. Use the securityContext principal this will be anonymous incase of non-secure
3. Let Anonymous user create the threads which is ok

Code is simpler we just depend on the SecurityContext session and not make any conditions who can and who cannot create threads
Anyone who is trying out and want to showcase the capabilities of OpenMetadata get to show the threads and comments. Even though they may be tagged as Anonymous. This will help them to demo and test features.

Reply:
@harshach Just to clarify, this patch doesn't restrict anonymous users from creating or getting threads. But I agree that we should not be conditionally checking if the user exists based on whether security is enabled or not. I have opened #2865 to update user reference to EntityReference and I will introduce a new user "anonymous" as part of that patch and remove this conditional check.